### ISSUE TITLE
Can't set a numeric value with versions of Puppet < 4.0.0

### ISSUE THREAD

**Issue by glarizza** [puppetlabs/puppetlabs-hocon#11]
Given the following manifest:

```puppet
  hocon_setting { 'exact text setting':
      ensure => present,
      path => '/tmp/foo.conf',
      setting => 'foo',
      value => 14,
      type => 'number',
    }
```

You cannot set this numeric value on versions of Puppet < 4.0.0 without using the future parser (using Puppet 3.7.4 in this test)

```
└(~/src/puppetlabs-hocon)▷ envpuppet puppet apply test.pp
Notice: Compiled catalog for satori.local in environment production in 0.19 seconds
Error: Parameter value failed on Hocon_setting[exact text setting]: Validate method failed for class value: Type specified as 'number' but was String at /Users/glarizza/src/puppetlabs-hocon/test.pp:7
Wrapped exception:
Validate method failed for class value: Type specified as 'number' but was String
```

This should be able to be fixed by modifying the validate block in the `value` type to something like this:

```ruby
        when 'number'
          # Puppet stringifies numerics in versions of Puppet < 4.0.0
          # Account for this in the type
          begin
            numeric_as_string = Integer(value)
          rescue ArgumentError
            numeric_as_string = false
          end
          unless (value.is_a?(Numeric) or numeric_as_string)
            raise "Type specified as 'number' but was #{value.class}"
          end
```

...and then munging the value:

```ruby
    munge do |value|
      if value.is_a?(String) and @resource[:type] == 'number'
        value = Integer(value)
      end
    end
```

The problem then is that it's not setting correctly within the value setter method.  I've not traced the code well enough to determine exactly WHY, but wanted to file the issue to get some eyes on it and make a sanity check...

**Comment by glarizza** [comment]
PR:  https://github.com/puppetlabs/puppetlabs-hocon/pull/12

**Comment by glarizza** [comment]
Merged in #12
